src/browser/api/window.js

```diff
@@ -1164,15 +1165,48 @@ Window.setWindowPreloadState = function(identity, payload) {
     const { uuid, name } = identity;
     const { url, state, allDone } = payload;
     const updateTopic = allDone ? 'preload-state-changed' : 'preload-state-changing';
-    let { preloadState } = Window.wrap(uuid, name);
-
-    // Single preload script state change
-    if (!allDone) {
-        preloadState = preloadState.find(e => e.url === url);
-        preloadState.state = state;
+    const frameInfo = coreState.getInfoByUuidFrame(identity);
+    let openfinWindow;
+    if (frameInfo.entityType === 'iframe') {
+        openfinWindow = Window.wrap(frameInfo.parent.uuid, frameInfo.parent.name);
+    } else {
+        openfinWindow = Window.wrap(uuid, name);
     }
 
-    ofEvents.emit(route.window(updateTopic, uuid, name), { name, uuid, preloadState });
+    if (openfinWindow) {
```

Comment:
can this be

```
if (!openfinWindow) {
   return log.writeToLog('info', `setWindowPreloadState missing openfinWindow ${uuid} ${name}`);
}
```

so that we don't nest the whole block of code with a tab?